docs/projects/m2k/index.rst

```diff
@@ -0,0 +1,616 @@
+.. _m2k:
+
+M2K HDL Project
+===============================================================================
+
+Overview
+-------------------------------------------------------------------------------
+
+The :adi:`ADALM2000 <ADALM2000>` (M2K), is built around the XC7Z010 Xilinx Zynq FPGA.
+
+Booting is done from a 32MB QSPI FLASH, connected directly to the PS7 block.
+
+The system memory is 512MB RAM, which is also used for the file system. The DDR
+interface is a 16 bits interface running at 500 MHz. The frequency was chosen so
+that a single PLL runs inside of the PS7 block generating all the required PS7
+clocks, consuming minimum power.
+
+The FPGA drives 16 GPIOs for the logic analyzer, 2 GPIOs for triggers, a 14 pin
+interface to the :adi:`AD9963`'s ADC and a 14 pin interface to the :adi:`AD9963`'s
+DAC. It also has several generic GPIOs, an SPI and an I2C interface for
+configuring the board.
+
+The control GPIOs and the SPI are driven by the PS7 block in order to minimize
+the PL resource usage.
+
+The I2C is implemented by an AXI I2C IP because the PS7 I2C has several
+limitations, which makes it not suitable for this project.
+
+The oscilloscope related logic transfers data from the :adi:`AD9963` chip with
+the :adi:`AD9963` TRX interface goes through the decimation block, the analog
+triggering block, the history FIFO and streams data to memory using the DMA.
+
+The analog waveform generator transfers data from the memory using two DMAs goes
+through the interpolation block and ends with the :adi:`AD9963` TX interface,
+which transfers data to the :adi:`AD9963` chip. Two DMAs are used in order to
+have different orders of magnitude for the frequency of the two AWG channels,
+without needing to transfer a very large number of samples.
+
+The logic analyzer and pattern generator logic are mostly implemented in the
+same IP to which a history FIFO and DMAs are added.
+
+Supported boards
+-------------------------------------------------------------------------------
+
+- :adi:`ADALM2000 <ADALM2000>`
+
+Supported devices
+-------------------------------------------------------------------------------
+
+- :adi:`AD9363`
+
+Block design
+-------------------------------------------------------------------------------
+
+Detailed description
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. collapsible:: PS7
+
+   The PS7 hardware block is the main processing block of the system, having an
+   ARM processor running at 500MHz. It is running a Linux operating system.
+
+   In order to minimize the power utilization for the design, the PS7 block is
+   the only block generating clocks in the system. The clock configuration is
```

Comment:
```suggestion
   To minimize the power utilization for the design, the PS7 block is
   the only block generating clocks in the system. The clock configuration is
```
\+ "the only block which generates clocks" or "the only clock-generating block"?